workers/agent/agent.go

```diff
@@ -454,9 +455,76 @@ func (a *Agent) scheduleWork() {
 
 		// Let us try to start the plugin and receive all implemented jobs
 		if err = a.scheduler.SetPipelineJobs(pipeline); err != nil {
-			gaia.Cfg.Logger.Error("cannot get pipeline jobs", "error", err.Error(), "pipelinerun", pipelineRunPB)
-			reschedulePipeline()
-			return
+			if strings.Contains(err.Error(), "exec format error") {
+				gaia.Cfg.Logger.Info("pipeline in a different format than worker; attempting to rebuild...")
+				err = nil
+				// Try rebuilding the pipeline...
+				if err := os.Remove(pipelineFullPath); err != nil {
+					gaia.Cfg.Logger.Error("failed to remove pipeline binary", "error", err.Error(), "pipelinerun", pipelineRunPB)
+					reschedulePipeline()
+					return
+				}
+				pCreate := &gaia.CreatePipeline{}
+				repo, err := a.client.GetGitRepo(ctx, &pb.PipelineID{Id: int64(pipeline.ID)})
+				if err != nil {
+					gaia.Cfg.Logger.Error("failed to get pipeline information", "error", err.Error(), "pipelinerun", pipelineRunPB)
+					reschedulePipeline()
+					return
+				}
+				pCreate.Pipeline = *pipeline
+				// Unfortunately, since pb.GitRepo has extra gRPC fields on it
+				// we can't use gaia.GitRepo(repo) here to convert immediately.
+				gitRepo := gaia.GitRepo{}
+				gitRepo.Username = repo.Username
+				gitRepo.Password = repo.Password
+				pk := gaia.PrivateKey{}
+				pk.Password = repo.PrivateKey.Password
+				pk.Username = repo.PrivateKey.Username
+				pk.Key = repo.PrivateKey.Key
+				gitRepo.PrivateKey = pk
+				gitRepo.URL = repo.Url
+				gitRepo.SelectedBranch = repo.SelectedBranch
+				pCreate.Pipeline.Repo = &gitRepo
+				//gp.CreatePipeline(pCreate)
+				//if pCreate.StatusType == gaia.CreatePipelineFailed {
+				//	gaia.Cfg.Logger.Error("cannot create pipeline", "output", pCreate.Output, "pipelinerun", pipelineRunPB)
+				//	reschedulePipeline()
+				//	return
+				//}
+
+				pb := gp.NewBuildPipeline(pipeline.Type)
```

Comment:
You can actually use `CreatePipeline` here. Worker also uses the storage service to store the pipeline run status updates.

Reply:
Interesting. I had an error when I tried that which said, nope. :O Huh. It might have been another problem. That would be cool, cause I think CreatePipeline does the gitCheckout and I wouldn't have to export these two functions.

Reply:
Ah you can actually see me trying out CreatePipeline below which is commented out because it did not work. I'll look into it some more.